trello-backup.py

```diff
@@ -1,127 +1,153 @@
-
-	for board in boards.json():
-		if ORGANIZATION_IDS and (not board["idOrganization"] or not board["idOrganization"] in ORGANIZATION_IDS):
-			continue
-
-		print(u"    - {0} ({1})".format(board["name"], board["id"]))
-		boardContents = requests.get(API_URL + "boards/" + board["id"], data=boardPayload)
-		with io.open(OUTPUT_DIRECTORY + u'/{0}_'.format(board["name"].replace("/","-")) + epoch_time + '.json', 'w', encoding='utf8') as file:
-			args = dict( sort_keys=True, indent=4) if PRETTY_PRINT else dict()
-			data = json.dumps(boardContents.json(), ensure_ascii=False, **args)
-			file.write(unicode(data))
-
-def get_organization_ids(ORGANIZATION_NAMES):
-	selected_organizations = []
-	for organization in ORGANIZATION_NAMES.split(','):
-		selected_organizations.append(organization.strip())
-	organization_ids = []
-
-	# Parameters to get a list of organizations
-	organizationsPayload = {
-		'key':API_KEY,
-		'token':TOKEN,
-	}
-
-	organizations = requests.get(API_URL + "members/my/organizations", data=organizationsPayload)
-	if len(organizations.json()) <= 0:
-		print('No organizations found.')
-	else:
-		for organization in organizations.json():
-			if organization["name"] in selected_organizations:
-				organization_ids.append(organization["id"])
-	return organization_ids
-
-
-if __name__ == '__main__':
-	main()
+import logging
+import argparse
+
+API_URL = "https://api.trello.com/1/"
+
+
+def get_organization_ids(ORGANIZATION_NAMES, API_KEY, TOKEN):
+    """
+    Translates Org Name to ID
+    """
+    selected_organizations = []
+    for organization in ORGANIZATION_NAMES.split(','):
+        selected_organizations.append(organization.strip())
+    organization_ids = []
+
+    # Parameters to get a list of organizations
+    organizationsPayload = {
+        'key': '{}'.format(API_KEY),
+        'token': '{}'.format(TOKEN),
+    }
+
+    organizations = requests.get(API_URL + "members/me/organizations", params=organizationsPayload)
+    if len(organizations.json()) <= 0:
+        logger.error('No organizations found.')
+    else:
+        logger.info('Finding all Organizations.')
+        for organization in organizations.json():
+            if organization["name"].lower() in selected_organizations:
+                organization_ids.append(organization["id"])
+                logger.debug('Org Added: {}'.format(organization["id"]))
+    return organization_ids
+
+if __name__ == "__main__":
+
+    # parse args
+    parser = argparse.ArgumentParser(description="Trello Backup")
+    parser.add_argument("-k", "--key", help="Trello API Key")
+    parser.add_argument("-t", "--oauth-token", help="Trello OAuth Token")
+    parser.add_argument("-d", "--directory", required=True, help="Directory to save JSON files, created if non exist")
+    parser.add_argument("-l", "--log", help="Logging level - info|INFO|WARNING|ERROR|CRITICAL")
+    parser.add_argument("-a", "--appname", required=True, help="Trello App Name")
```

Comment:
Might want to standardise all the arguments to a single format `-` or `camelCasing`, e.g. oauth-token uses `-`, appname should probably be app-name